File: quizmastime-frontend/src/app/models/prize.ts
export interface Prize {
  id?: number;
  name: string;
  description?: string;
  imageUrl?: string;
  assignedUserId?: number;
  assignedUserFirstName?: string;
  assignedUserLastName?: string;
}

export interface PrizeAssignment {
  prizeId: number;
  userId?: number;
}

File: quizmastime-backend/src/main/java/com/quizmastime/backend/model/Prize.java
package com.quizmastime.backend.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "prize")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Prize {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "Prize name is required")
    @Column(nullable = false)
    private String name;

    @Column(length = 1000)
    private String description;

    @Column
    private String imageUrl;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "assigned_user_id")
    private User assignedUser;
}
